world.js

```diff
@@ -94,173 +89,183 @@ const _bindState = (state, dynamic) => {
 };
 _bindState(states.static, false);
 _bindState(states.dynamic, true);
+
+function makeid(length) {
+  let result = '';
+  const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+  for (let i = 0; i < length; i++) {
+    result += characters.charAt(Math.floor(Math.random() * characters.length));
+  }
+  return result;
+}
+
+function buffVec(buffer, vec) {
+  buffer[0] = vec.x;
+  buffer[1] = vec.y;
+  buffer[2] = vec.z;
+
+  if (vec.w !== undefined) {
+    buffer[3] = vec.w;
+  }
+
+  return buffer;
+}
+
+function buffArgs(buffer, ...args) {
```

Comment:
I could be wrong, but I suspect `...` is going to be less efficient than using an array.

If this is core loop function it's probably worth that change.